add-on/src/lib/ipfs-request.js

```diff
@@ -450,6 +450,16 @@ function createRequestModifier (getState, dnslinkResolver, ipfsPathValidator, ru
       const state = getState()
       if (!state.active) return
       if (request.statusCode === 200) return // finish if no error to recover from
+
+      // Seamlessly fix canonical link when DNSLink breaks ipfs.io/blog/*
+      /// https://github.com/ipfs/blog/issues/360
+      if (request.type === 'main_frame' &&
+          request.statusCode === 404 &&
+          request.url.includes('/ipns/ipfs.io/blog')) {
```

Comment:
Would be safer to use regex (or other way) to ensure is included directly following FQDN.